src/constructs/rds/instance.ts

```diff
@@ -1,12 +1,34 @@
-import { Fn } from "aws-cdk-lib";
+import { Fn, Tags } from "aws-cdk-lib";
 import { InstanceType } from "aws-cdk-lib/aws-ec2";
 import { DatabaseInstance } from "aws-cdk-lib/aws-rds";
 import type { DatabaseInstanceProps } from "aws-cdk-lib/aws-rds";
 import { GuAppAwareConstruct } from "../../utils/mixin/app-aware-construct";
 import type { AppIdentity, GuStack } from "../core";
 
-export interface GuDatabaseInstanceProps extends Omit<DatabaseInstanceProps, "instanceType">, AppIdentity {
+export interface OptIn {
+  enabled: true;
+}
+
+// Using Pick here means that we get the AWS documentation for these properties rather than needing to define it ourselves
+export interface OptOut extends Pick<DatabaseInstanceProps, "backupRetention" | "preferredBackupWindow"> {
+  enabled: false;
+  /**
+   * We recommend using DevX Backups where possible. If it is not suitable for your use-case please document
+   * this here so that we can understand why this is switched off when performing security audits.
+   */
+  optOutReason: string;
+}
+
+export interface GuDatabaseInstanceProps
+  extends Omit<DatabaseInstanceProps, "instanceType" | "backupRetention" | "preferredBackupWindow">,
+    AppIdentity {
   instanceType: string;
```

Comment:
It seems a shame that we are removing the type from `instanceType` and making it a `string` instead - I'd like to change this but it'd be a breaking change and is not linked to the PR, so have left it as is for now.

Reply:
Agree on both counts.